SOLIDHomework.Core/Program.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using SOLIDHomework.Core;
+using SOLIDHomework.Core.Builder;
+using SOLIDHomework.Core.Interfaces;
+using SOLIDHomework.Core.Model;
+using SOLIDHomework.Core.Services;
+
+namespace SOLIDHomework
+{
+    public class Program
+    {
+        //Entry point to program
+        //You don't have to change logic there
+        //Tip: that is good place for composition root
+        static void Main(string[] args)
+        {
+            var paymentService = new PaymentService();
+            var inventoryService = new InventoryService();
+            var notificationService = new NotificationService();
+
+            
+            // Create a new service collection
+            var serviceCollection = new ServiceCollection();
+            // Register services
+            ConfigureServices(serviceCollection);
+            // Build the service provider
+            var serviceProvider = serviceCollection.BuildServiceProvider();
+            // Resolve the logger and use it
+            var logger = serviceProvider.GetService<ILogger>();
+            var _paymentService = serviceProvider.GetService<IPaymentService>();
+            var _notificationService = serviceProvider.GetService<INotificationService>();
+            var _IInventoryService = serviceProvider.GetService<IInventoryService>();
+            OrderService orderService = new OrderService(paymentService, inventoryService, notificationService, logger);
```

Comment:
You don't need to pass values in the OrderService, they should be injected by DI container

Reply:
interface for OrderService